### PR TITLE
Set up CI pipeline

### PR DESCRIPTION
**Pull request by alayek**
Set up a CI pipeline in Github Actions to run the following:

- TypeScript checks
- Linting

We can add testing later on with Jest.

### CODE REVIEW

**Comment by alayek** [comment]
I have read and accept the CLA.

**Comment by crimelabs786** [comment]
@cla-bot check now

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  pull_request:
+    branches: [main] # run on PRs opened against `main`
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: true
+      matrix:
+        node: [16]
```
Comment (crimelabs786) [inline .github/workflows/ci.yml:13]: Have we validated that it indeed runs against Node v16 as in `.nvmrc` file?

@alayek can you please add a step that prints output of `node --version`?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        uses: actions/checkout@v2
+        with:
+          persist-credentials: false
+          fetch-depth: 0
+
+      - name: Yarn Install
+        run: yarn install --frozen-lockfile
+
+      - name: TypeScript Compilation
+        run: yarn tsc --noEmit
```
Comment (crimelabs786) [inline .github/workflows/ci.yml:26]: Add a script in `package.json` that runs type check. Something like `yarn type-check`.

File: .vscode/extensions.json [hunk 3]
```diff
@@ -2,6 +2,7 @@
   "recommendations": [
     "dbaeumer.vscode-eslint",
     "esbenp.prettier-vscode",
-    "bradlc.vscode-tailwindcss"
+    "bradlc.vscode-tailwindcss",
+    "redhat.vscode-yaml"
```
Comment (crimelabs786) [inline .vscode/extensions.json:6]: Why's this extension needed?

File: package.json [hunk 4]
```diff
@@ -7,7 +7,8 @@
     "build": "next build",
     "start": "next start",
     "add-git-hooks": "husky install",
-    "postinstall": "is-ci || husky install"
+    "postinstall": "is-ci || husky install",
+    "lint": "yarn eslint . && yarn stylelint **/*.css"
```
Comment (crimelabs786) [inline package.json:11]: Maybe break it into two pieces, one for JS linting, another for style linting. Then this command can be invoked ergonomically.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (alayek) [reply .github/workflows/ci.yml:13]: You were right. I've now added an extra read step for `.nvmrc`

File: .github/workflows/ci.yml (hunk 2 above)
Comment (alayek) [reply .github/workflows/ci.yml]: Fixed.

File: .vscode/extensions.json (hunk 3 above)
Comment (alayek) [reply .vscode/extensions.json:6]: It gives auto-complete for GitHub's YAML syntax, and intellisense around documentation, from YAML Schema store.

File: package.json (hunk 4 above)
Comment (alayek) [reply package.json]: Fixed.

**Comment by alayek** [comment]
@crimelabs786 have addressed your comments, please review once.

Adding testing is pending. Will add it as we add unit and integration testing.

**Review by crimelabs786 — APPROVED** [review]

**Pull request merged**